File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.cpp
class MyStack {
public:
    queue<int>q;
    int t;
    MyStack() {
        
    }
    
    void push(int x) {
        q.push(x);
        t = x;
    }
    
    int pop() {
        queue<int>q2;
        while(q.size() > 1) {
            t = q.front(); q.pop();
            q2.push(t);
        }
        int x = q.front(); q.pop();
        q = q2;
        return x;
    }
    
    int top() {
        return t;
    }
    
    bool empty() {
        return q.empty();
    }
};

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack* obj = new MyStack();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->top();
 * bool param_4 = obj->empty();
 */

File: 0225-implement-stack-using-queues/0225-implement-stack-using-queues.java
class MyStack {
    Queue<Integer> q1 = new LinkedList<>();
    int temp;
    public MyStack() {
        
    }
    
    public void push(int x) {
        q1.add(x);
        temp = x;
    }
    
    public int pop() {
        Queue<Integer> q2 = new LinkedList<>();
        while(q1.size() > 1) {
            temp = q1.peek();
            q2.add(q1.poll());
        }
        int x = q1.poll();
        q1 = q2;
        return x;
    }
    
    public int top() {
        return temp;
    }
    
    public boolean empty() {
        return q1.isEmpty();
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */